src/main/java/edu/kis/powp/jobs2d/factories/RectFactory.java

```diff
@@ -0,0 +1,21 @@
+package edu.kis.powp.jobs2d.factories;
+
+import edu.kis.powp.jobs2d.commands.ComplexCommand;
+import edu.kis.powp.jobs2d.commands.OperateToCommand;
+import edu.kis.powp.jobs2d.commands.SetPositionCommand;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class RectFactory implements ActionListener {
```

Comment:
Factory shouldn't implement ActionListener, instead you should have methods returning new ComplexCommands